data/data/azure/master/master.tf

```diff
@@ -38,6 +38,7 @@ resource "azurerm_virtual_machine" "master" {
   count                 = var.instance_count
   name                  = "${var.cluster_id}-master-${count.index}"
   location              = var.region
+  zones                 = compact([var.availability_zones[count.index]])
```

Comment:
This should be `element()` to correctly map Masters to AZs when the count of masters is more than count of AZs

Reply:
Looking at https://github.com/openshift/installer/pull/1949/commits/9e807295cfb1e411b6fe71b3512811e2aa4f322d#diff-e1983977011640174c1c393e26c9ac96R93 we should be fine.

Reply:
I need to update the name to match machine api objects though